File: packages/java/src/test/java/com/ailabtools/sdk/GeneratedSurfaceTest.java
package com.ailabtools.sdk;

import static org.junit.jupiter.api.Assertions.assertEquals;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import org.junit.jupiter.api.Test;

final class GeneratedSurfaceTest {
    @Test
    void generatedApiSurfaceContainsAllDocumentedEndpoints() throws IOException {
        Path api = Path.of("src", "main", "java", "com", "ailabtools", "sdk", "api");
        long count;
        try (var files = Files.list(api)) {
            count = files.filter(path -> path.getFileName().toString().endsWith("Api.java"))
                    .flatMap(path -> {
                        try {
                            return Files.lines(path);
                        } catch (IOException error) {
                            throw new IllegalStateException(error);
                        }
                    })
                    .filter(line -> line.contains("requester.request("))
                    .count();
        }
        assertEquals(93, count);
    }
}

File: packages/kotlin/src/test/kotlin/com/ailabtools/sdk/GeneratedSurfaceTest.kt
package com.ailabtools.sdk

import java.nio.file.Files
import java.nio.file.Path
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test

internal class GeneratedSurfaceTest {
    @Test
    fun `generated API surface contains every documented endpoint`() {
        val apiRoot = Path.of("src", "main", "kotlin", "com", "ailabtools", "sdk", "api")
        val apiSource = Files.walk(apiRoot).use { paths ->
            paths.iterator().asSequence()
                .filter { it.fileName.toString().endsWith("Api.kt") }
                .map { Files.readString(it) }
                .joinToString("\n")
        }
        val endpointCalls = Regex("""requester\.request\(""").findAll(apiSource).count()
        val uniquePaths = Regex(""""/api/[^"]+"""")
            .findAll(apiSource)
            .map { it.value }
            .toSet()

        assertEquals(93, endpointCalls)
        assertEquals(93, uniquePaths.size)
        assertTrue(apiSource.contains("suspend fun"))
    }
}
